internal/api/attestationconfigapi/reporter.go

```diff
@@ -0,0 +1,197 @@
+)
+
+// cachedVersionsSubDir is the subdirectory in the bucket where the cached versions are stored.
+const cachedVersionsSubDir = "cached-versions"
+
+// versionWindowSize defines the number of versions to be considered for the latest version. Each week 5 versions are uploaded for each node of the verify cluster.
+const versionWindowSize = 15
+
+var reportVersionDir = path.Join(attestationURLPath, variant.AzureSEVSNP{}.String(), cachedVersionsSubDir)
+
+// UpdateLatestAzureSEVSNPVersion reports the given version, checks the reported version values
+// and updates the latest version of the Azure SEVSNP in the API if there is an update .
+// force can be used to force an update of the latest API version regardless of the cache reporting.
+func (c Client) UpdateLatestAzureSEVSNPVersion(ctx context.Context, inputVersion,
+	latestAPIVersion AzureSEVSNPVersion, now time.Time, force bool,
+) error {
+	if err := c.reportAzureSEVSNPVersion(ctx, inputVersion, now); err != nil {
+		return fmt.Errorf("reporting version: %w", err)
+	}
+	if force {
+		if err := c.uploadAzureSEVSNPVersion(ctx, inputVersion, now); err != nil {
+			return fmt.Errorf("force uploading version: %w", err)
+		}
+		return nil
+	}
+	versionDates, err := c.listReportedVersions(ctx)
+	if err != nil {
+		return fmt.Errorf("list reported versions: %w", err)
+	}
+	if len(versionDates) < c.cacheWindowSize {
+		c.s3Client.Logger.Infof("Skipping version update since found only %d out of expected reported versions.", len(versionDates), c.cacheWindowSize)
+		return nil
+	}
+	minVersion, minDate, err := c.findMinVersion(ctx, versionDates)
+	if err != nil {
+		return fmt.Errorf("get minimal version: %w", err)
+	}
+	c.s3Client.Logger.Infof("Found minimal version: %+v with date: %s", *minVersion, minDate)
+	shouldUpdateAPI, err := isInputNewerThanOtherVersion(*minVersion, latestAPIVersion)
+	if err == nil && shouldUpdateAPI {
+		c.s3Client.Logger.Infof("Input version: %+v is newer than latest API version: %+v", *minVersion, latestAPIVersion)
+		t, err := time.Parse(VersionFormat, minDate)
+		if err != nil {
+			return fmt.Errorf("parsing date: %w", err)
+		}
+		if err := c.uploadAzureSEVSNPVersion(ctx, *minVersion, t); err != nil {
+			return fmt.Errorf("uploading version: %w", err)
+		}
+		c.s3Client.Logger.Infof("Successfully uploaded new Azure SEV-SNP version: %+v", *minVersion)
+		return nil
+	}
+	c.s3Client.Logger.Infof("Input version: %+v is not newer than latest API version: %+v", *minVersion, latestAPIVersion)
+	return nil
+}
+
+// SetCacheVersionSize sets a custom number of versions to be considered for the latest version.
+func (c *Client) SetCacheVersionSize(size int) {
+	c.cacheWindowSize = size
+}
+
+// reportAzureSEVSNPVersion uploads the latest observed version numbers of the Azure SEVSNP. This version is used to later report the latest version numbers to the API.
+func (c Client) reportAzureSEVSNPVersion(ctx context.Context, version AzureSEVSNPVersion, date time.Time) error {
```

Comment:
```suggestion
func (c Client) cacheAzureSEVSNPVersion(ctx context.Context, version AzureSEVSNPVersion, date time.Time) error {
```
We are talking about caching versions in multiple places already.